### ISSUE TITLE
Can't drop external objects

### ISSUE THREAD

**Issue by kyrias** [olirice/alembic_utils#73]
I'm currently trying to hack together a `ReplaceableObject` for creating and dropping composite types, and at least until it's cleaned up a bit I'm keeping it in our application repository.

For getting `alembic_utils` to _create_ the objects of my type I just have to append my own entity types to the `register_entities` argument of `register_entities`.  But when checking for whether it should drop anything, `alembic_utils` calls `collect_subclasses` again and only checks the returned entity types, meaning that my type will be ignored for drop checks.

https://github.com/olirice/alembic_utils/blob/bdaa374d27c597ba79cd1d0b56ea5d3e87404a07/src/alembic_utils/replaceable_entity.py#L321-L325

Why is it doing this double check there?  I can't think of any reason for it not to just loop over `allowed_entity_types`, because it seems very strange to _create_ objects of other types but not be able to drop them.

**Comment by kyrias** [comment]
For the record, I'm currently working around this with this:

```python
def collect_subclasses(module, class_):
    found = experimental.collect_subclasses(module, class_)
    return found + entity_types

replaceable_entity.collect_subclasses = collect_subclasses
```

But I'd rather not monkey patch things unless it's really necessary.

**Comment by olirice** [comment]
I updated `collect_subclasses` so that it also returns the contents of `class_.__subclasses__()`. That includes all imported subclasses. 

To register your internal subclass to work with drops, make sure it is imported at the top of `env.py` 

ref: ae8a5db60fac0d34d713cdb780fd2d7b1ae1d3ab

that update is available in `alembic_utils==0.7.5`

let me know if you have any trouble getting it working

**Comment by kyrias** [comment]
That was fast!  It seems to work great, thanks!
